gstd/src/async_runtime/locks.rs

```diff
@@ -0,0 +1,66 @@
+//! Wait duration registry
+use crate::{async_runtime, exec, prelude::BTreeMap, MessageId};
+
+/// Wait locks.
+#[derive(Debug, PartialEq, Eq)]
+pub(crate) enum Lock {
+    For(u32),
+    NoMore(u32),
+}
+
+impl Lock {
+    /// Call wait functions by the lock type.
+    pub fn wait(&self) {
+        match self {
+            Lock::For(d) => exec::wait_for(*d),
+            Lock::NoMore(d) => exec::wait_no_more(*d),
+        }
+    }
+}
+
+impl Default for Lock {
+    fn default() -> Self {
+        unsafe { Lock::NoMore(crate::config::DEFAULT_WAIT_NO_MORE_DURATION) }
+    }
+}
+
+/// Wait trait for async sending messages.
+pub trait Wait: Sized {
+    /// Delays handling for given specific amount of blocks.
+    fn no_more(self, duration: u32) -> Self {
+        async_runtime::locks().insert(crate::msg::id(), Lock::NoMore(duration));
```

Comment:
Such realisation of functions is bugged:
```rust
let f = msg::send_bytes_for_reply(to, [], 0)
    .expect("send message failed")
    .no_more(10);

let _ = msg::send_bytes_for_reply(to, [], 0)
    .expect("send message failed")
    .no_more(5);
  
let _res = f.await; // this will wait no more 10, instead of proper 5 
```

Reply:
To solve this, we should set lock type on polling of future.

Reply:
Also here might be a problem of polling several futures: `FuturesUnordered/join!(..)/etc.` - need to find the general solution and provide tests (probably deny `StreamExt` for custom waiting)